src/main/java/com/redhat/rhjmc/containerjfr/net/BasicAuthManager.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.rhjmc.containerjfr.net;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+import com.redhat.rhjmc.containerjfr.core.sys.FileSystem;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+
+class BasicAuthManager extends AbstractAuthManager {
+
+    static final String USER_PROPERTIES_FILENAME = "container-jfr-users.properties";
+
+    private final FileSystem fs;
+    private final Properties users;
+    private volatile boolean configLoaded = false;
+
+    // TODO salted hashes
+    BasicAuthManager(Logger logger, FileSystem fs) {
+        super(logger);
+        this.fs = fs;
+        this.users = new Properties();
+    }
+
+    @Override
+    public Future<Boolean> validateToken(Supplier<String> tokenProvider) {
```

Comment:
Would it be overkill to operate on character arrays for anything dealing with the incoming credentials and nulling out the password characters immediately after use?

Reply:
Er, overwriting with zeroes rather than nulling in this case.

Reply:
Actually it's coming in from a library anyways, it's probably overkill.

Reply:
It would be doable, but I wouldn't consider that to be too much of a security improvement, if that's the concern. It narrows the timing window where the password is in memory, but not considerably so, and doesn't do anything about the other places in memory where the password had to pass through first - for example, the webserver's HTTP header we parsed it out of. 

And if an attacker has the ability to read your ContainerJFR credentials out of JVM memory, then they probably already have access to whatever targets your ContainerJFR would have connected to and more...